tfhe/src/safe_serialization.rs

```diff
@@ -236,22 +229,35 @@ impl SerializationConfig {
         object: &T,
         mut writer: impl std::io::Write,
     ) -> bincode::Result<()> {
-        let options = bincode::DefaultOptions::new()
-            .with_fixint_encoding()
-            .with_limit(0);
+        let options = bincode::DefaultOptions::new().with_fixint_encoding();
 
         let header = self.create_header::<T>();
-        options
-            .with_limit(self.header_length_limit())
-            .serialize_into(&mut writer, &header)?;
+        let header_size = options.serialized_size(&header)?;
 
-        match self.versioned {
-            SerializationVersioningMode::Versioned { .. } => options
-                .with_limit(self.serialized_size_limit)
-                .serialize_into(&mut writer, &object.versionize())?,
-            SerializationVersioningMode::Unversioned { .. } => options
-                .with_limit(self.serialized_size_limit)
-                .serialize_into(&mut writer, &object)?,
+        if let Some(size_limit) = self.serialized_size_limit {
+            options
+                .with_limit(size_limit)
+                .serialize_into(&mut writer, &header)?;
+
+            match self.versioned {
+                SerializationVersioningMode::Versioned { .. } => options
+                    .with_limit(size_limit - header_size)
```

Comment:
` options.with_limit(size_limit - header_size)` could be gotten out of the match